web/src/main/java/org/cbioportal/web/config/WebServletContextListener.java

```diff
@@ -0,0 +1,113 @@
+package org.cbioportal.web.config;
+
+import org.apache.commons.lang.StringUtils;
+import org.cbioportal.proxy.ProxyController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Value;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import java.io.*;
+import java.util.Properties;
+
+/**
+ * Created by Hongxin Zhang on 11/15/19.
+ */
+public class WebServletContextListener implements ServletContextListener, InitializingBean {
+    private Logger LOG = LoggerFactory.getLogger(WebServletContextListener.class);
+
+    private Boolean showOncokb;
+    
+    private String oncokbToken;
+    
+    private String oncokbPublicApiUrl;
+
+    @Override
+    public void contextDestroyed(ServletContextEvent arg0) {
+        System.out.println("ServletContextListener destroyed");
+    }
+
+    //Run this before web application is started
+    @Override
+    public void contextInitialized(ServletContextEvent arg0) {
+        Properties properties = loadProperties(getResourceStream("portal.properties"));
+        
+        this.showOncokb = Boolean.parseBoolean(properties.getProperty("show.oncokb"));
+        this.oncokbToken = properties.getProperty("oncokb.token");
+        this.oncokbPublicApiUrl = properties.getProperty("oncokb.public_api.url");
+
+        if (this.showOncokb) {
+            checkOncokbInfo();
+        }
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+    }
+
+    private void checkOncokbInfo() {
+        System.out.println("------------------------");
+        if(StringUtils.isEmpty(this.oncokbToken)) {
+            // oncokb.org is deprecated, www.oncokb.org should be used
+            System.out.println("OncoKB requires token to access the latest data. Please review oncokb terms(https://www.oncokb.org/terms) and register an account accordingly. For now, we will redirect the requests to a legacy instance which will not accept any new data changes. Thank you.");
+            LOG.warn("OncoKB requires token to access the latest data. Please review oncokb terms(https://www.oncokb.org/terms) and register an account accordingly. For now, we will redirect the requests to a legacy instance which will not accept any new data changes. Thank you.");
```

Comment:
No sure what I did wrong, but the LOG.warn does not print out the message. Only the `System.out.println` works.

Reply:
might need to change log4j properties to print WARN to stdout. With default props I think it outputs to the log file. We might want to change the default `log4j.properties`. This is an example for my debug config:

```
# Change INFO to DEBUG, if you want to see debugging info on underlying libraries we use.
log4j.rootLogger=INFO, a

# Change INFO to DEBUG, if you want see debugging info on our packages and spring security packages.
log4j.category.org.mskcc=INFO
log4j.logger.org.springframework.security=INFO

# Use the JVM option, e.g.: "java -DPORTAL_HOME=/pathto/portal_homedir",
# or - "java -DPORTAL_HOME=$PORTAL_HOME", where PORTAL_HOME is shell (environment) variable.

## IMPORTANT - THRESHOLD SHOULD NOT BE DEBUG FOR PRODUCTION, CREDENTIALS CAN BE DISPLAYED!
log4j.appender.a=org.apache.log4j.ConsoleAppender
log4j.appender.a.Target=System.out
log4j.appender.a.layout=org.apache.log4j.PatternLayout

log4j.logger.org.cbioportal.persistence.mybatis.StudyMapper=TRACE
log4j.logger.org.cbioportal.persistence.mybatis=DEBUG
src/main/resources/log4j.properties.DEBUG (END)
```

Reply:
@inodb this prints out the warning not log.debug. So technically in the default which prints INFO would also include the WARN.